test/parallel/test-net-connect-buffer.js

```diff
@@ -20,68 +20,63 @@
 // USE OR OTHER DEALINGS IN THE SOFTWARE.
 
 'use strict';
-require('../common');
+const common = require('../common');
 const assert = require('assert');
 const net = require('net');
 
-let dataWritten = false;
-let connectHappened = false;
-
-const tcp = net.Server(function(s) {
+const tcp = net.Server(common.mustCall((s) => {
   tcp.close();
 
   console.log('tcp server connection');
 
   let buf = '';
+  s.setEncoding('utf8');
   s.on('data', function(d) {
     buf += d;
   });
 
   s.on('end', function() {
-    console.error('SERVER: end', buf.toString());
+    console.error('SERVER: end', buf);
     assert.strictEqual(buf, "L'État, c'est moi");
-    console.log('tcp socket disconnect');
     s.end();
   });
+}));
 
-  s.on('error', function(e) {
-    console.log(`tcp server-side error: ${e.message}`);
-    process.exit(1);
-  });
-});
-
-tcp.listen(0, function() {
+tcp.listen(0, common.mustCall(function() {
   const socket = net.Stream({ highWaterMark: 0 });
 
-  console.log('Connecting to socket ');
-
-  socket.connect(this.address().port, function() {
-    console.log('socket connected');
-    connectHappened = true;
-  });
+  console.log('Connecting to socket');
 
-  console.log(`connecting = ${socket.connecting}`);
+  let connected = false;
+  socket.connect(this.address().port, common.mustCall(() => connected = true));
 
+  assert.strictEqual(socket.connecting, true);
   assert.strictEqual('opening', socket.readyState);
```

Comment:
Nit: now that you are here, can you please swap the arguments? There are more below.

Reply:
Yup, done!